src/Base/Stream.cpp

```diff
@@ -32,200 +32,123 @@
 # include <string>
 # include <cstdio>
 # include <cstring>
-#ifdef __GNUC__
-# include <stdint.h>
+# include <climits>
 #endif
-#endif
-
+#include <iomanip>
 #include "Stream.h"
-#include "Swap.h"
 #include <CXX/Objects.hxx>
 #include <Python.h>
 
 using namespace Base;
 
-Stream::Stream() : _swap(false)
-{
-}
-
-Stream::~Stream()
-{
-}
-
-Stream::ByteOrder Stream::byteOrder() const
-{
-    return _swap ? BigEndian : LittleEndian;
-}
-
-void Stream::setByteOrder(ByteOrder bo)
-{
-    _swap = (bo == BigEndian);
-}
-
-OutputStream::OutputStream(std::ostream &rout) : _out(rout)
-{
-}
-
-OutputStream::~OutputStream()
-{
-}
-
-OutputStream& OutputStream::operator << (bool b)
-{
-    _out.write((const char*)&b, sizeof(bool));
-    return *this;
-}
-
-OutputStream& OutputStream::operator << (int8_t ch)
-{
-    _out.write((const char*)&ch, sizeof(int8_t));
-    return *this;
-}
-
-OutputStream& OutputStream::operator << (uint8_t uch)
-{
-    _out.write((const char*)&uch, sizeof(uint8_t));
-    return *this;
-}
-
-OutputStream& OutputStream::operator << (int16_t s)
-{
-    if (_swap) SwapEndian<int16_t>(s);
-    _out.write((const char*)&s, sizeof(int16_t));
-    return *this;
-}
-
-OutputStream& OutputStream::operator << (uint16_t us)
+OutputStream::OutputStream(std::ostream &rout, bool binary)
+    : _out(rout), _binary(binary)
 {
-    if (_swap) SwapEndian<uint16_t>(us);
-    _out.write((const char*)&us, sizeof(uint16_t));
-    return *this;
-}
-
-OutputStream& OutputStream::operator << (int32_t i)
-{
-    if (_swap) SwapEndian<int32_t>(i);
-    _out.write((const char*)&i, sizeof(int32_t));
-    return *this;
+    if(!_binary)
+        _out << std::setprecision(std::numeric_limits<float>::digits10 + 1);

```

Comment:
Maybe use `max_digits10` instead?
```        
_out << std::setprecision(std::numeric_limits<float>::max_digits10);
```

Reply:
Yeah, after C++11, you should use that instead. There are quite a few other places that `digits10` is used. I think we can change them together in a dedicated PR.

Reply:
Fortunately only nine other places, so easy to review. If you'll take care of this one I can look at the others.